compiler/noirc_evaluator/src/acir/call/intrinsics/slice_ops.rs

```diff
@@ -24,25 +27,136 @@ impl Context<'_> {
+
+            // 1. Convert the elements-to-push into flattened acir_var and at the same time
+            // push_back corresponding dummy zero values to the AcirValues slice.
+            for (elem, ssa_typ) in elements_to_push.iter().zip(slice_types.to_vec()) {
+                let element = self.convert_value(*elem, dfg);
+                element_size += super::arrays::flattened_value_size(&element);
+                match element {
+                    AcirValue::Var(acir_var, acir_type) => {
+                        new_slice.push_back(AcirValue::Var(zero, acir_type));
+                        elements_var.push(acir_var);
+                    }
+                    AcirValue::Array(vector) => {
+                        let zero_value = self.array_zero_value(&ssa_typ)?;
+                        new_slice.push_back(zero_value);
+                        for acir_value in vector {
+                            let acir_vars = self.flatten(&acir_value)?;
+                            elements_var.extend(acir_vars);
+                        }
+                    }
+                    AcirValue::DynamicArray(_) => {
+                        unimplemented!("pushing a dynamic array into a slice is not yet supported");
+                    }
+                }
+            }
+
+            // The actual flattened size of new_slice after the dummy push_back
+            let new_slice_array = AcirValue::Array(new_slice);
+            let len = super::arrays::flattened_value_size(&new_slice_array);
+
+            // 2. Copy the slice into an AcirDynamicArray
+            // Generates the element_type_sizes array
+            let element_type_sizes =
+                if super::arrays::array_has_constant_element_size(&slice_typ).is_none() {
+                    Some(self.init_element_type_sizes_array(
+                        &slice_typ,
+                        slice_contents,
+                        Some(&new_slice_array),
+                        dfg,
+                    )?)
+                } else {
+                    None
+                };
+
+            // The block ID for the new slice is the one for the resulting slice
+            let block_id = self.block_id(result_ids[1]);
+            self.initialize_array(block_id, len, Some(new_slice_array))?;
+            let flattened_dynamic_array =
+                AcirDynamicArray { block_id, len, value_types, element_type_sizes };
+
+            // 3. Write to the dynamic array
+
+            // 3.1 Computes the flatten_idx where to write into the dynamic array:
+            // Use element_type_size if it exists; convert the user index (slice_length) into the AcirValues index,
+            // and then flatten it with element_type_size
+            let mut flatten_idx = if let Some(element_type_sizes) = element_type_sizes {
+                let predicate_index = self
+                    .acir_context
+                    .mul_var(slice_length, self.current_side_effects_enabled_var)?;
```

Comment:
Is the side effects var needed here? I find it strange because no other slice intrinsic seems to rely on it.